src/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -159,10 +165,16 @@ function discoverController($scope, config, courier, $route, $window, Notifier,
   $scope.uiState = $state.makeStateful('uiState');
 
   function getStateDefaults() {
+    const defaultColumns = config.get('defaultColumns').slice();
+
+    const timeFieldName = $scope.indexPattern.timeFieldName;
+    if (timeFieldName) {
+      defaultColumns.unshift(timeFieldName);
```

Comment:
Right, so this is no worse than the current behavior. And having `@timestamp` in the `defaultColumns` would make it show up in non-timestamped index patterns as well, which is no good.